weblate/api/tests.py

```diff
@@ -211,6 +212,46 @@ def test_create(self):
         )
         self.assertEqual(Project.objects.count(), 2)
 
+    def test_create_with_source_language(self):
+        self.do_request(
+            "api:project-list",
+            method="post",
+            code=403,
+            format="json",
+            request={
+                "name": "API project",
+                "slug": "api-project",
+                "web": "https://weblate.org/",
+                "source_language": {
+                    "code": "ru",
+                    "name": "Russian",
+                    "direction": "ltr",
+                },
+            },
+        )
+        response = self.do_request(
+            "api:project-list",
+            method="post",
+            code=201,
+            superuser=True,
+            format="json",
+            request={
+                "name": "API project",
+                "slug": "api-project",
+                "web": "https://weblate.org/",
+                "source_language": {
+                    "code": "ru",
+                    "name": "Russian",
+                    "direction": "ltr",
+                },
+            },
+        )
+        self.assertEqual(Project.objects.count(), 2)
+        self.assertEqual(response.data["source_language"]["code"], "ru")
+        self.assertEqual(
+            Project.objects.get(slug="api-project").source_language.code, "ru"
+        )
```

Comment:
Can you please also add test for error handling? Trying to create project with non existing source language.